web/html/src/manager/admin/payg-shared/sshData/payg-ssh-data-edit.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from "react";
+
+import { paygProperties } from "manager/admin/payg/payg";
+import useLifecyclePaygActionsApi from "manager/admin/payg-shared/api/payg-actions-api";
+import PaygSshDataForm from "manager/admin/payg-shared/sshData/payg-ssh-data-form";
+import PaygSshDataView from "manager/admin/payg-shared/sshData/payg-ssh-data-view";
+
+import CreatorPanel from "components/panels/CreatorPanel";
+import { showErrorToastr, showSuccessToastr } from "components/toastr";
+import { Loading } from "components/utils";
+
+type Props = {
+  paygSshData: paygProperties;
+  paygId: string;
+  isInstance: boolean;
+  labelPrefix: string;
+  onChange: Function;
+  editing?: boolean;
+};
+
+const PaygSshDataEdit = (props: Props) => {
+  const { onAction, cancelAction, isLoading } = useLifecyclePaygActionsApi();
+
+  const saveAction = ({ item, closeDialog, setErrors }) => {
+    onAction(item, "update", props.paygId)
+      .then((data) => {
+        closeDialog();
+        showSuccessToastr(t("Pay-as-you-go properties updated successfully"));
+        props.onChange(data);
+      })
+      .catch((error) => {
+        setErrors(error.errors);
+        if (error.messages) {
+          showErrorToastr(error.messages, { autoHide: false });
+        } else {
+          showErrorToastr(error, { autoHide: false });
+        }
+      });
+  };
+  return (
+    <div id={props.labelPrefix + "-panel-wrapper"}>
+      <CreatorPanel
+        id={(props.isInstance ? "bastion_" : "") + "connection"}
+        creatingText={t("Edit " + props.labelPrefix)}
+        panelLevel="2"
+        title={props.labelPrefix + " SSH connection"}
```

Comment:
This should probably also be translated.

Reply:
Done